packages/flutter/lib/src/material/expansion_tile.dart

```diff
@@ -669,14 +669,14 @@ class _ExpansionTileState extends State<ExpansionTile> with SingleTickerProvider
   }
 
   Widget? _buildLeadingIcon(BuildContext context) {
-    if (_effectiveAffinity(widget.controlAffinity) != ListTileControlAffinity.leading) {
+    if (_effectiveAffinity(widget.controlAffinity ?? ListTileTheme.of(context).controlAffinity) != ListTileControlAffinity.leading) {
```

Comment:
It looks like _effectiveAffinity could be `getEffectiveAffinity()`:
```dart
  ListTileControlAffinity _getEffectiveAffinity() {
    final ListTileControlAffinity affinity =
      widget.controlAffinity ?? ListTileTheme.of(context).controlAffinity ?? ListTileControlAffinity.trailing;
    return affinity == ListTileControlAffinity.platform ? ListTileControlAffinity.trailing : affinity;
  }
```

And then this expression could be `(_getEffectiveAffinity() != ListTileControlAffinity.leading)`

I just typed this in without really checking it. The idea is that the common code could be factored out and the effectiveAffinity computation could be simpler.

Reply:
resolved, see [commit](https://github.com/flutter/flutter/pull/146630/commits/eb31a043f8b4031cc63b3632221d98141c7aad3e)